EIPS/eip-7873.md

```diff
@@ -155,21 +155,39 @@ At the start of the block in which [EIP-3540](./eip-3540.md) activates, set the
 
 ```solidity
 {
-    /// Takes [tx_initcode_hash][salt][init_data] as input,
-    /// creates contract and returns the address or failure otherwise
+    /// Takes [tx_initcode_hash][salt][unsafe_flag][init_data] as input,
+    /// creates contract and returns the address or failure otherwise.
 
-    // init_data.length can be 0, but the first 2 words are mandatory
-    let size := calldatasize()
-    if lt(size, 64) { revert(0, 0) }
+    /// tx_initcode_hash and salt are 32 bytes wide, unsafe_flag is 1 byte wide, init_data can be any width.
+    /// If unsafe_flag is 0x00, `CALLER` is included in the salt and impacts the target address of the created account.
+    /// If unsafe_flag is non-zero, `CALLER` is excluded. Deployment by any sender will result in same target address. 
+    /// Deployments using non-zero unsafe_flag are susceptible to front-running.
 
-    calldatacopy(0, 0, size)  // copy tx_initcode_hash, salt and init_data to memory to hash
-    let final_salt := keccak256(0, size)
+    // init_data.length can be 0, but the first 65 bytes are mandatory
+    let size := calldatasize()
+    if lt(size, 65) { revert(0, 0) }
+
+    // copy tx_initcode_hash, salt, unsafe_flag and init_data to memory to hash.
+    // note that the unsafe_flag is included to prevent pre-image attacks on the two paths to calculate salt below.
+    calldatacopy(0, 0, size)
+
+    // mask out the 31 bytes that follow the flag in input data
+    let unsafe_flag := calldataload(64) & (0xff << 248)
+
+    if iszero(unsafe_flag) {
+        // Include sender into salt just after init_data
+        mstore(msize(), caller())
+        let final_salt := keccak256(0, size + 32)
```

Comment:
I think there is a bug here: `msize()` is rounded up to multiple of 32, but `init_data` may not be a multiple, so this code results in some padding between `init_data` and sender, and since the hash uses `size` it does not actually include the full address in the final salt.